File: java/sui-entity/src/main/java/ru/sui/suientity/config/Autoconfiguration.java
package ru.sui.suientity.config;

import org.springframework.boot.autoconfigure.domain.EntityScan;
import org.springframework.cache.annotation.EnableCaching;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.jpa.repository.config.EnableJpaAuditing;
import org.springframework.data.jpa.repository.config.EnableJpaRepositories;

@ComponentScan(basePackages = "ru.sui.suientity")
@EntityScan("ru.sui.suientity.entity")
@EnableJpaRepositories("ru.sui.suientity.repository")
@EnableJpaAuditing
@EnableCaching
@Configuration
public class Autoconfiguration {}

File: java/sui-security/src/main/java/ru/smsoft/sui/suisecurity/config/Autoconfiguration.kt
package ru.smsoft.sui.suisecurity.config

import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity

@Configuration
@EnableWebSecurity
@EnableGlobalMethodSecurity(securedEnabled = true, jsr250Enabled = true, prePostEnabled = true)
@ComponentScan("ru.smsoft.sui.suisecurity")
class Autoconfiguration
